### PR TITLE
Adding bicep file for webapp managed mysql quickstart

### PR DESCRIPTION
**Pull request by juliakm**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding a bicep deploy file for the [Build a Web App on Linux with Azure database for MySQL sample](https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.web/webapp-linux-managed-mysql). I'd like to use reference this update in this [existing quickstart for Azure DevOps Pipelines](https://docs.microsoft.com/azure/devops/pipelines/apps/cd/azure/deploy-arm-template?view=azure-devops).

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by juliakm** [comment]
#sign-off

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    endIpAddress: '255.255.255.255'
+  }
+  dependsOn: [
+    serverName_databaseName
+  ]
+}
+
+resource serverName_databaseName 'Microsoft.DBforMySQL/servers/databases@2017-12-01' = {
+  parent: serverName
+  name: '${databaseName}'
```
Comment (jbpaux) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep]: interpolation is not needed

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        }
+      ]
+    }
+    name: siteName
+    serverFarmId: hostingPlanName.id
+  }
+}
+
+resource serverName 'Microsoft.DBforMySQL/servers@2017-12-01' = {
+  name: serverName_var
```
Comment (jbpaux) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep]: resource should be something like `mysqlserver` and the `name` attribute be `serverName` (was caused by decompile but should be cleaned afterwards)

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    name: 'S1'
+  }
+  kind: 'linux'
+  properties: {
+    name: hostingPlanName_var
+    reserved: true
+  }
+}
+
+resource siteName_resource 'Microsoft.Web/sites@2020-06-01' = {
```
Comment (jbpaux) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep]: resource should be something like `site` (was caused by decompile but should be cleaned afterwards)

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Azure database for mySQL sku family')
+param databaseskuFamily string = 'Gen5'
+
+var databaseName = 'database${uniqueString(resourceGroup().id)}'
+var serverName_var = 'mysql-${uniqueString(resourceGroup().id)}'
+var hostingPlanName_var = 'hpn-${uniqueString(resourceGroup().id)}'
```
Comment (jbpaux) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep]: Rename variables to omit `_var`

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+resource hostingPlanName 'Microsoft.Web/serverfarms@2020-06-01' = {
+  name: hostingPlanName_var
```
Comment (jbpaux) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep]: resource should be something like `hostingPlan` and the `name` attribute be `hostingPlanName` (was caused by decompile but should be cleaned afterwards)

**Review by jbpaux — CHANGES_REQUESTED** [review]
I added a few comments here and you must also compile the bicep file

**Comment by jbpaux** [comment]
You need to compile the bicep to azuredeploy.json and commit the file.

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by juliakm** [comment]
@bmoore-msft I think I've addressed the feedback and pipelines errors. Is there an example I can look at for a template without bpa errors? I'm not sure how to fix those.

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by juliakm** [comment]
@bmoore-msft Can you take a look at this again?

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep [hunk 6]
```diff
@@ -0,0 +1,149 @@
+@description('main.bicep')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:1]: fix description

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+param siteName string = '${uniqueString(resourceGroup().name)}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:2]: remove string interpolation and use resourceGroup().id as the seed

```param siteName string = uniqueString(resourceGroup().id)```

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+])
+param mysqlVersion string = '5.7'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Azure database for mySQL sku family')
+param databaseskuFamily string = 'Gen5'
+
+var databaseName = '${uniqueString(resourceGroup().id)}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:65]: in the original sample, these 3 vars (and the siteName param) had static prefixes to make them easier to read - is removing that intentional?  If so, remove the string interpolation on all 3...  If not, add back the prefixes here and on the site name param

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+          name: 'defaultConnection'
+          connectionString: 'Database=${databaseName};Data Source=${mysqlserver.properties.fullyQualifiedDomainName};User Id=${administratorLogin}@${mysqlserverName};Password=${administratorLoginPassword}'
+          type: 'MySql'
+        }
+      ]
+    }
+    serverFarmId: hostingPlan.id
+    httpsOnly: true
+  }     
+   identity: {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:102]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+resource serverName_AllowAzureIPs 'Microsoft.DBforMySQL/servers/firewallrules@2017-12-01' = {
+  parent: mysqlserver
+  name: 'AllowAzureIPs'
+  properties: {
+    startIpAddress: '0.0.0.0'
+    endIpAddress: '255.255.255.255'
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:137]: remove dependsOn

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep (hunk 10 above)
Comment (juliakm) [reply quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep]: Removed now.

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep (hunk 9 above)
Comment (juliakm) [reply quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep]: I think that this section was the only one with the incorrect order. Should be resolved now.

File: quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep (hunk 8 above)
Comment (juliakm) [reply quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep]: It was in response to a different comment. I've removed the string interpolation on all three.

**Pull request merged**